segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/StreamSegmentNotSealedException.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.contracts;
+
+public class StreamSegmentNotSealedException extends StreamSegmentException {
+    /**
+     *
+     */
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * Creates a new instance of the StreamSegmentNotSealedException class.
+     *
+     * @param streamSegmentName The name of the StreamSegment that is not sealed.
+     */
+    public StreamSegmentNotSealedException(String streamSegmentName) {
```

Comment:
Is this only used with txns?

Reply:
It's used whenever a Segment needs to be sealed but it's not. In our case, it's when a merge/concat source is not sealed, so yes, currently only used with transactions.

Reply:
ok